src/zarr/storage/_common.py

```diff
@@ -163,24 +163,15 @@ async def get(
             prototype = default_buffer_prototype()
         return await self.store.get(self.path, prototype=prototype, byte_range=byte_range)
 
-    async def set(self, value: Buffer, byte_range: ByteRequest | None = None) -> None:
+    async def set(self, value: Buffer) -> None:
```

Comment:
I'm not sure I understand this change. Local and memory storage does support writing bytes into an existing file, and this would be very important for efficient writes to uncompressed, sharded arrays.